### DISCUSSION TITLE
m4a files not showing up

### DISCUSSION THREAD

**Discussion by cweiske** [gerbera/gerbera#3182] [Q&A] [answered]
I have a folder with 10 .opus and 6 .m4a files, and only the .opus files are visible in the virtual directory structure. The .m4a ones are not showing up in the web interface > Database > Artists > ...

This is a self-compiled version of gerbera, v2.2.0-10-gacfd1ac0.

What could I be missing that m4a are not supported?

The gerbera log shows that both opus and m4a files get imported (my import script logs that).

Compile info
-------------
WITH_NPUPNP=OFF
WITH_MAGIC=ON
WITH_MYSQL=YES
WITH_CURL=ON
WITH_INOTIFY=ON
WITH_JS=YES
WITH_TAGLIB=ON
WITH_AVCODEC=OFF
WITH_FFMPEGTHUMBNAILER=OFF
WITH_EXIF=ON
WITH_EXIV2=ON
WITH_MATROSKA=ON
WITH_WAVPACK=OFF
WITH_SYSTEMD=ON
WITH_LASTFM=OFF
WITH_DEBUG=YES
WITH_DEBUG_OPTIONS=YES
WITH_TESTS=OFF

**Comment by KarlStraussberger** [comment, marked as answer]
Avcodec aka FFMPEG might help here. Also it higly depends on the version of taglib. They are only handling a small set of m4a containers correctly. I attached my current (somewhat stuck) experiments to get more file accepted.

[taglib.patch](https://github.com/user-attachments/files/16599459/taglib.patch)

**Reply by cweiske** [reply]
Compiling with `-DWITH_AVCODEC=YES` makes the files appear in the virtual directory layout.
The cover images are missing, but it's better than nothing.

**Reply by KarlStraussberger** [reply]
Yes, because for ffmpeg we currently do not support coverart.
